src/components/TodoContainer.jsx

```diff
@@ -4,39 +4,104 @@ import TodoList from "./TodoList.jsx";
+            data.records.sort((objectA, objectB) => {
+                return sortByFieldName(isAsc, objectA.fields[sortFieldName], objectB.fields[sortFieldName]);
+            });
+            console.log(data.records)
+
+            const todos = data.records.map((record) => {
+                return {
+                    id: record.id,
+                    title: record.fields.title,
+                    createdTime: record.createdTime
+                }
+            })
+            console.log('todos', todos);
+
+            setTodoList(todos);
+            if (todos) {
+                setIsLoading(false);
+            }
+        };
+        const {fieldName, isAsc} = parseSortArguments(sort);
+
+        getData(isAsc, fieldName);
     }, [])
 
+    // To not query API for sorting
+    // TODO: move this to API if pagination is a case
+    useEffect(() => {
+        const {fieldName, isAsc} = parseSortArguments(sort);
+        const newTodoList = [...todoList].sort((objectA, objectB) => {
+            return sortByFieldName(isAsc, objectA[fieldName], objectB[fieldName]);
+        });
+        setTodoList(newTodoList);
+    }, [sort])
+
     useEffect(() => {
         const strinifiedTodoList = JSON.stringify(todoList);
         if (isLoading === false) {
             localStorage.setItem("savedTodoList", strinifiedTodoList);
         }
     }, [todoList]);
 
-    const addTodo = (newTodo) => {
-        createData(newTodo.title, setTitleError, setTodoList, todoList)
+    const addTodo = async (newTodo) => {
+        const {createdTodo, error} = await createData(newTodo.title)
+        if (error) {
+            setTitleError(error);
+        } else {
+            const newTodoList = [...todoList, createdTodo];
+            const {fieldName, isAsc} = parseSortArguments(sort);
+            newTodoList.sort((objectA, objectB) => {
+                return sortByFieldName(isAsc, objectA[fieldName], objectB[fieldName]);
+            });
+            setTodoList(newTodoList);
+        }
     };
 
-    const removeTodo = (id) => {
-        removeData(id, setTodoList, todoList)
+    const removeTodo = async (id) => {
+        const oldTodoList = [...todoList]
+        setTodoList(todoList.filter((todo) => todo.id !== id));
+
+        const isDeleted = await removeData(id, setTodoList, todoList);
+
+        if (!isDeleted) {
+            setTodoList(oldTodoList);
+        }
     }
 
+    const sortChange = (event) => {
+        setSort(event.target.value);
+    };
+
     return (
-        <Fragment>
+        <>
             <h1>Todo List</h1>
             <AddTodoForm onAddTodo={addTodo} titleError={titleError}/>
+            <select onChange={sortChange} value={sort}>
+                <option value="title-asc">Title a-z</option>
+                <option value="title-desc">Title z-a</option>
+                <option value="createdTime-asc">Created Time a-z</option>
+                <option value="createdTime-desc">Created Time z-a</option>
```

Comment:
With this approach, you may benefit from using lookup objects (or enums in TypeScript) rather than text. I see elsewhere you're parsing this text + splitting based on dashes. This is good approach but I can see where a system like this could expand and you'd have an increasingly hard time keeping track of all the valid text strings. If you go with a lookup object approach, you'll have the advantage of auto-completion and decrease confusion from developers having a hard time understanding the rules you've implicitly set up around option values.